### ISSUE TITLE
custom objective for lambdarank/ranking

### ISSUE THREAD

**Issue by xin-jin** [Microsoft/LightGBM#1896]
Is custom objective function supported for ranking models? I would like to tweak the lambdarank loss a little bit. Since the loss function needs to know the group information, what would the loss function signature be?

Thanks!

**Comment by guolinke** [comment]
refer to https://github.com/Microsoft/LightGBM/blob/3ad9cba0321a8ac2e5f625da6dcc6f7c1516a750/examples/python-guide/advanced_example.py#L140-L145 
you can use the `train_data.get_group()`

**Comment by 1a1a11a** [comment]
is the function signature `def loglikelihood(preds, train_data)` or `def loglikelihood(y_true, pred)`? Because here https://towardsdatascience.com/custom-loss-functions-for-gradient-boosting-f79c1b40466d says the later and when I print the two input types are `<class 'numpy.ndarray'> <class 'numpy.ndarray'>`.

**Comment by 1a1a11a** [comment]
Maybe the function for different interfaces (sklearn and LightGBM) are different?

**Comment by guolinke** [comment]
@1a1a11a yeah, they are different.

**Comment by siamakz** [comment]
Hi @guolinke 
I am also trying to define a custom objective for ranking problem. Can you please elaborate more on how it can be done? I am just starting to work with LightGBM. Thanks!
